src/Migrator.EF6.Tools/DotnetToolDispatcher.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.PlatformAbstractions;
+using NuGet.Frameworks;
+
+namespace Microsoft.Extensions.Internal
+{
+	internal static class DotnetToolDispatcher
```

Comment:
Csproj does not offer exactly the same functionality as the project.json did. One major change is the possible inclusion of source files.

With project.json it was possible to complie “external” package source files into a project and therefore access internal classes. That’s exactly what you and Microsoft did to include DotnetToolDispatcher.cs:
"Microsoft.Extensions.DotnetToolDispatcher.Sources": {
	"type": "build",
	"version": "1.0.0-*"
}
With (old) new csproj that is not possible anymore. You have to physically add the class to your project, like I did, or to compile the source file into the project.
<ItemGroup>
	<Compile Include="..\..\shared\Microsoft.Extensions.CommandLineUtils.Sources\**\*.cs" />
</ItemGroup>